### PR TITLE
Remove unused ROS1 code and launch files

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
This PR removed ROS1 related code in conjunction with https://github.com/usdot-fhwa-stol/carma-utils/pull/234 and https://github.com/usdot-fhwa-stol/carma-config/pull/383

- **launch files** that are no longer used
- **guidance_plugin_validator** which is not needed as all plugins are ROS2
- **ns3_adapter** this code was factored out to a different repository long time ago.

## Related GitHub Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2079

## Related Jira Key
[CAR-146](https://usdot-carma.atlassian.net/browse/ARC-146)

## Motivation and Context
Migration to Foxy

## How Has This Been Tested?
Built carma-platform on local VM

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

[CAR-146]: https://usdot-carma.atlassian.net/browse/CAR-146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMD

### CODE REVIEW

File: docker/checkout.bash [hunk 1]
```diff
@@ -40,7 +40,7 @@ done
 cd "${dir}"/src
 
 git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-msgs.git --branch "${BRANCH}"
-git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-utils.git --branch "${BRANCH}"
+git clone --depth=1 https://github.com/usdot-fhwa-stol/carma-utils.git --branch arc-146-remove-ros1
```
Comment (adev4a) [inline docker/checkout.bash:43]: This can be reverted to the argument

File: carma/launch/carma_src.launch [hunk 2]
```diff
@@ -33,19 +33,12 @@
...
-  <arg name="cuda" default="true" doc="cuda flag for vision_darknet_detect (true if cuda is available)"/>
-  <arg name="enable_guidance_plugin_validator" default="false" doc="Flag indicating whether the Guidance Plugin Validator node will actively validate guidance strategic, tactical, and control plugins"/>
-  <arg name="strategic_plugins_to_validate" default="[]" doc="List of String: Guidance Strategic Plugins that will be validated by the Guidance Plugin Validator Node if enabled"/>
-  <arg name="tactical_plugins_to_validate" default="[]" doc="List of String: Guidance Tactical Plugins that will be validated by the Guidance Plugin Validator Node if enabled"/>
-  <arg name="control_plugins_to_validate" default="[]" doc="List of String: Guidance Control Plugins that will be validated by the Guidance Plugin Validator Node if enabled"/>
-  <arg name="combined_lidar_frame" default="velodyne" doc="Frame of the lidar(s)"/>
 
 	<!-- Startup Drivers With Main CARMA System -->
   <arg name="launch_drivers" default="true" doc="True if drivers are to be launched with the CARMA Platform, overrides mock_drivers arg if false"/>
```
Comment (adev4a) [inline carma/launch/carma_src.launch:38]: This argument can also be removed, along with drivers.launch file it looks like nothing is being launched there

File: carma/launch/carma_src.launch [hunk 3]
```diff
@@ -85,26 +71,12 @@
     </include>
   </group>
 
-
```
Comment (adev4a) [inline carma/launch/carma_src.launch:88]: Its not letting me add a comment on the correct line, but the "launch_drivers" and "vehicle_calibration_dir" args don't need to be passed to hardware_interface.launch. Please also remove reference to those arguments in  hardware_interface.launch.

**Comment by adev4a** [comment]
Inlanecruising currently specifies a dependency on trajectory_utils instead of trajectory_utils_ros2 which should be updated https://github.com/usdot-fhwa-stol/carma-platform/blob/b0fcfc7a64c382f410789f154f7be7e629eb4aa3/inlanecruising_plugin/package.xml#L40 since that package is being removed as part of https://github.com/usdot-fhwa-stol/carma-utils/pull/234

File: carma/launch/carma_src.launch (hunk 3 above)
Comment (MishkaMN) [reply carma/launch/carma_src.launch:88]: removed all references of drivers.launch and launch_drivers

File: docker/checkout.bash (hunk 1 above)
Comment (MishkaMN) [reply docker/checkout.bash:43]: I can remove it before I merge, since the build will fail

File: carma/launch/hardware_interface.launch [hunk 4]
```diff
@@ -15,37 +15,14 @@
...
-
-    <!-- Driver Launch File if Using Actual Drivers -->
-    <include file="$(find carma)/launch/drivers.launch">
-      <arg name="mock_drivers" value="$(arg mock_drivers)" />
-      <arg name="vehicle_calibration_dir" value="$(arg vehicle_calibration_dir)"/>
-    </include>
-  </group>
-	
   <!-- Temporary node to get a ROS2 accel topic into the rosbag -->
   <node pkg="topic_tools" type="relay" name="relay_vehicle_accel" args="/hardware_interface/as/velocity_accel_cov /hardware_interface/velocity_accel_cov"/>
```
Comment (adev4a) [inline carma/launch/hardware_interface.launch:27]: Is this node still required if we are defaulting to recording rosbags in ros2?

File: inlanecruising_plugin/package.xml [hunk 5]
```diff
@@ -36,10 +36,9 @@
   <depend>carma_wm</depend>
   <depend>basic_autonomy</depend>
   <depend>tf2</depend>
-  <depend>tf2_geometry_msgs</depend> 
-  <depend>trajectory_utils</depend>
```
Comment (adev4a) [inline inlanecruising_plugin/package.xml:40]: References should also be removed from headers

File: carma/launch/hardware_interface.launch (hunk 4 above)
Comment (MishkaMN) [reply carma/launch/hardware_interface.launch:27]: I don't think we are defaulting to run rosbags in ros2 just yet. Suntrax demo for example is scheduled to record the data in ROS1 this year. We will move to ROS2 when the developers have set tools and scripts to analyze them.

File: inlanecruising_plugin/package.xml (hunk 5 above)
Comment (MishkaMN) [reply inlanecruising_plugin/package.xml:40]: removed the references

**Review by adev4a — APPROVED** [review]

**Comment by adev4a** [comment]
Looks good to me! Please revert the branch in the checkout script before merging

**Review by adev4a — APPROVED** [review]

**Pull request merged**